### PR TITLE
Cleanup gemspec, rakefile and upgrade dependencies

### PR DESCRIPTION
**Pull request by knapo**
The main reason I creating this PR is that our fossa scanner complains that `redgreen` gem (which is a dev dependence of `jewler` gem) doesn't have a licence specified. Since both `redgreen` and `jewler` are not longer being maintained, `certificate_authority` is the gem which requires some care :)

What this PR does is:
- Change travis config to test against recent ruby versions
- Stop using `jewler`, which is not longer maintained (gem can be simply released with `rake build` + `rake release`)
- Upgrade and cleanup gemspec to follow most recent style
- Cleanup Rakefile
- Fix `constant ::Fixnum is deprecated` warning
- Add `.rspec` file and remove individual helper requires.

Once the PR is merged, I think it's time to release 1.0.0 :)

### CODE REVIEW

**Comment by knapo** [comment]
@cchandler Hope, you'll find a while to review and merge this PR. Thanks!

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+---
+sudo: false
 language: ruby
+cache: bundler
 rvm:
-  - 1.9.3
```
Comment (xiazek) [inline .travis.yml]: it could be some people still use 1.9.3 (?)

Comment (knapo) [reply .travis.yml]: Well, it ended its life and is not supported by the ruby-core team any longer and does not receive any fixes. I don't see a reason to release a new version supporting dead ruby versions. If someone still uses it, he probably doesn't upgrade anything including dependent gems ;) 

The only maintained versions are 2.4 (until end of this month), 2.5, 2.6, 2.7.

Comment (xiazek) [reply .travis.yml]: Good point. I'm for the removal from CI also then.

**Comment by cchandler** [comment]
Hey, thanks for looking into this! Most of it is long overdue 😅. I'll see if I can find some time this week to merge it in.

**Comment by knapo** [comment]
@cchandler just a friendly reminder :)

**Comment by cchandler** [comment]
Things are kinda returning to normal for me. Taking a look this afternoon.

**Comment by knapo** [comment]
Many thanks! If you can release a new version with the changes, it would be
awesome!

**Comment by cchandler** [comment]
Thank you so much for the time and contribution! I've bumped the version to 1.0.0 (on master) and pushed to rubygems. The last time I published this gem was apparently 2012 🤣

**Pull request merged**